Updated TTF file with the latest 4.0.3 version of the FontAwesome direct...

lucasiturbide opened the pull request:
...ly from the Website.

Updated the NSString category with the latest icon. A refactor was needed to update easyly the icon values and a better way to convert to FA string from an Enum value in HEX.

Deprecated unnecessary methods. Keeps the backwards compatibility of the Icon names.

The Enum has the same name convention than the CSS file on FA website, but keeps the older names for backward compatibility.

Fixed the Random icon generator in Demo App accordingly to the new enum based icons.

lucasiturbide committed:
Updated TTF file with the latest 4.0.3 version of the FontAwesome directly from the Website.
Updated the NSString category with the latest icon. A refactor was needed to update easyly the icon values and a better way to convert to FA string from an Enum value in HEX.

Deprecated unnecessary methods. Keeps the backwards compatibility of the Icon names.

The Enum has the same name convention than the CSS file on FA website, but keeps the older names for backward compatibility.

Fixed the Random icon generator in Demo App accordingly to the new enum based icons.

jessesquires commented:
thanks again @lucasiturbide !

using hex was a good idea - definitely much simpler. however, specifying each enum value makes this difficult to maintain. (but it is still much better than before).

would this work?

Don't specify enum values:

```
typedef NS_ENUM(NSUInteger, FAIcon){
    FAIconGlass,
    FAIconMusic,
    FAIconSearch,
    ....
}
```

Then simply use the hex format specifier:

```
+ (NSString *)fa_stringForFontAwesomeIcon:(FAIcon)icon {
    return [NSString stringWithFormat:@"%x", icon];
}
```

---

**Other issues:**
1. I'm not interested in backwards compatibility and deprecating methods. Anything not used should simply be removed. This project abides by semantic versioning. These changes will result in a new release, `4.0.0`.
2. We should follow the exact naming conventions from the web.
3. Travis is still broken unfortunately, but I'm pretty sure these changes break existing unit-tests. You should run the unit-tests and update accordingly.
4. Other comments are in-line

jessesquires commented on BButton/Classes/BButton.m:
```diff
@@ -257,7 +255,6 @@ - (void)setShouldShowDisabled:(BOOL)show
 - (void)didRecieveMemoryWarningNotification:(NSNotification *)notification
 {
     NSLog(@"%@ recieved %@", [BButton class], notification.name);
-    kFontAwesomeStrings = nil;
```
since this method does nothing now, please remove.

also remove where this notification observer is added and removed.

jessesquires commented on BButtonDemo/TWBSDemoViewController.m:
```diff
@@ -19,6 +19,10 @@ - (void)viewDidLoad
+    // This is to get a random icon from the Enum. Since we cannot get a random enum in Objective-C, we need to create an array of them to get a random value from a subset of the whole enum.
+    NSUInteger arrayOfIcons[] = {FASortAlphaAsc, FASortAlphaDesc, FASortAmountAsc, FASortAmountDesc, FASortNumericAsc, FASortNumericDesc, FAThumbsUp, FAThumbsDown, FAYoutubeSquare, FAYoutube, FAXing, FAXingSquare, FAYoutubePlay, FADropbox, FAStackOverflow, FAInstagram, FAFlickr, FAAdn, FABitbucket, FABitbucketSquare, FATumblr, FATumblrSquare, FALongArrowDown, FALongArrowUp, FALongArrowLeft, FALon ...
+    int sizeOfIconsArray = sizeof(arrayOfIcons)/sizeof(arrayOfIcons[0]);
+    
```
this isn't acceptable. 

however, if my previous suggestion regarding the use of format specifiers (`@"%x"`) works, then the original code will work.

lucasiturbide replied:
I spent a couple of hours trying to figure out  a way to do a random pick from the enums, but it seems that is not possible on Objective-C.
This is why I had to make an array of int values from the Enum values, and then I could use the random generator to pick one from that array. 
Anyway, this will be used only on this Demo, I don't see any use of "Get a random icon from FontAwesome".

jessesquires replied:
You're right, there isn't a use for a random icon. But I want the demo and tests to be easily maintainable. 
We can do this:
- The first  enum is `0xf000`, which is `61440`.
- The last enum is `0xf196`, which is `61846`.
- Using these bounds, we can generate a random number, then cast it as an `unsigned short`

Because of the enum values are not consecutive, we will occasionally end up with this strange junk character, but that's ok.

lucasiturbide commented:
For the enum values, the answer is no. That won't work, because in the enums if you don't assign a value, it will start from 0 to infinity, sequentially. That's why I assign the hex value for each value in the enum. I took the hex values from the CSS file, some several minutes with a text editor that has a regex find&replace utility. 
For the backward compatibility, if you want I can remove everything that is not necessary anymore instead of deprecating the methods, but imagine that someone wants to update the whole library to use the latest icons from FontAwesome and they have spread in all the code, all the methods with the old enum values. That will be a pain in the ass to update the library and use the latest values.
That's why I took the time to match the old enum values with the HEX codes. 
Regarding the naming convention, I couldn't use the exact names from the CSS because objective-c will not allow you to use a dash in the name of the values (like fa-foursquare). Again the magic regex did the work for me.
Let me know you thoughts about this.

lucasiturbide committed:
Removed unused method didRecieveMemoryWarningNotification

lucasiturbide committed:
Fixed Broken tests

lucasiturbide commented:
Hi Jesse, I think the build is failing because the test should run on iOS 7.0+ and not on 6.0 like travis is trying to run the tests. I don't know how to change it yet. Let me know if you can figure out the problem.

jessesquires commented:
hey @lucasiturbide - tests are running for iOS 7.

seems to be a known issue with travis. i tried tweaking the script more but it is still failing. (though everything works fine for me locally).

issue opened with travis here: https://github.com/travis-ci/travis-ci/issues/2179

jessesquires commented:
@lucasiturbide - issue fixed! details available on the thread in the issue noted above.

please update your PR with the latest from develop! :smiley:

lucasiturbide commented:
@jessesquires I did rebase and pushed again, I believe it's ready for merging.

jessesquires commented:
thanks for the update @lucasiturbide ! i have a few more changes and questions.

specifying the hex value for each icon is too much work, we can do this instead:

```
typedef NS_ENUM(unsigned short, FAIcon){
    FAIconGlass = 0xf000,
    FAIconMusic,
    FAIconSearch,
    FAIconEnvelope,
    FAIconHeart,
    FAIconStar,
    FAIconStarEmpty,
    FAIconUser,
    FAIconFilm,
    FAIconThLarge,
    FAIconTh,
    FAIconThList,
    FAIconOk,
    FAIconRemove,
    FAIconZoomIn,
    FAIconZoomOut = 0xf010,
    FAIconOff,
    FAIconSignal,
    .......
```

Unless there's a reason we shouldn't?

Normally, only having the first assignment would be ok (`FAIconGlass = 0xf000`) however, it appears that these are not consecutive values, thus the need to reset at `FAIconZoomOut` (which is `0xf010`, but would be `0xf00f` if these were consecutive). 

So we will have to explicitly set the value every few icons. (There doesn't seem to be a pattern.) I wonder why FontAwesome does this. :cry:

jessesquires commented on BButton/Classes/NSString+FontAwesome.h:
```diff
@@ -40,326 +40,725 @@ extern NSString * const kFontAwesomeFont;
- *  @return An array of all the font awesome icon strings.
- */
-+ (NSArray *)fa_allFontAwesomeStrings;
++ (NSArray *)fa_allFontAwesomeStrings __attribute__((deprecated("No needed anymore. Use Method fa_stringFromFontAwesomeForIcon:(FAIcon)icon instead.")));
```
let's remove this method completely.

jessesquires commented on BButton/Classes/NSString+FontAwesome.h:
```diff
@@ -40,326 +40,725 @@ extern NSString * const kFontAwesomeFont;
- */
-+ (NSString *)fa_stringFromFontAwesomeStrings:(NSArray *)strings
-                                      forIcon:(FAIcon)icon;
++ (NSString *)fa_stringFromFontAwesomeStrings:(NSArray *)strings forIcon:(FAIcon)icon __attribute__((deprecated("Use Method fa_stringFromFontAwesomeForIcon:(FAIcon)icon instead.")));
```
let's remove this method completely.

lucasiturbide committed:
Removed deprecated methods.
Changed the method to generate random values on the Demo App

lucasiturbide commented:
@jessesquires  I've removed the deprecated methods and fixed the random generated icons.
Regarding the values in the enum, I don't think that using the method of setting only the first value will be a good approach, since the values are not consecutive, and there are some duplicated values, which have the name convention that you were using. Maybe we can remove those values if you think that they are not needed anymore.

jessesquires commented:
@lucasiturbide - nice! :+1:  Sounds good about the enums, that's probably best. (I wish they were consecutive!)
1. Duplicates - yes let's remove them
2. Naming - let's follow exactly what FontAwesome does (i.e., `FAPencilSquare`, `FAPencilSquareO`, etc.)

Thanks for all of your hard work! :smile:

lucasiturbide committed:
Removed old values in the Enum definition.
Replaced old values with the new ones

lucasiturbide commented:
Hi @jessesquires, I've removed duplicated old entries and fixed the tests and demo with new values!

jessesquires commented:
hey @lucasiturbide - thanks so much! looks great!

jessesquires commented:
sorry for the delay, been super busy!
